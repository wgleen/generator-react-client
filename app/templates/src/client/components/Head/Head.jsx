import React from 'react'
import { Helmet } from 'react-helmet'

const Head = props => {
  return (
    <Helmet>
      <title>Testing environment</title>
    </Helmet>
  )
}

export default Head