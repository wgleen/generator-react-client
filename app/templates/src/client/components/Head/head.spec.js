import React from 'react'
import { mount } from 'enzyme'
import Head from './Head'
import Helmet from 'react-helmet'

describe('<Head />', () => {
  it('should render correctly', () => {
    let component = mount(
      <Head />
    )

    expect(component).toMatchSnapshot()
    
    component.unmount()
  })

  it('should contain title tag with text "Testing environment"', () => {
    let component = mount(
      <Head />
    )

    expect(Helmet.peek().title).toEqual('Testing environment')
    
    component.unmount()
  })
})