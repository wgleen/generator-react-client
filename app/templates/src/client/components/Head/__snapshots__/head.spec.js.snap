// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Head /> should render correctly 1`] = `
<Head>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <SideEffect(NullComponent)
      defer={true}
      encodeSpecialCharacters={true}
      title="Testing environment"
      titleAttributes={Object {}}
    >
      <NullComponent
        defer={true}
        encodeSpecialCharacters={true}
        title="Testing environment"
        titleAttributes={Object {}}
      />
    </SideEffect(NullComponent)>
  </HelmetWrapper>
</Head>
`;
